ghost/core/core/server/services/milestones/service.js

```diff
@@ -69,10 +72,33 @@ module.exports = {
     },
 
     /**
+     * @param {number} [customTimeout]
+     *
      * @returns {Promise<object>}
      */
-    async initAndRun() {
+    async initAndRun(customTimeout) {
+        /**
+        * @param {number} ms
+        * @returns {Promise<void>}
+        */
+        async function sleep(ms) {
+            return new Promise((resolve) => {
+                setTimeout(resolve, ms);
+            });
+        }
+
+        const timeOut = customTimeout || JOB_TIMEOUT;
+
+        const today = new Date();
+        const msNow = today.getMilliseconds();
+        const newMs = msNow + timeOut;
+        const jobDate = today.setMilliseconds(newMs);
+
         await this.init();
+
+        logging.info(`Running milestone emails job on ${new Date(jobDate).toString()}`);
+
+        await sleep(timeOut);
```

Comment:
I don't think we want the init method to wait for up to 3 days, I think it would be better to return from the function and have the code run in a timeout separately

Reply:
Switched to use a separate fn which returns a promise with the timeout and doesn't wait for it to be resolved. This should unblock anything that would prevent `boot` from finishing.